### PR TITLE
Support Federated Queries with Aliased Keys

### PR DESCRIPTION
**Pull request by CNAChino**
# What Changed
- Support Federated queries with aliased keys
- upgrade to graphql-java 17.5

Todo:

- [x] Add tests
- [ ] Add docs

### CODE REVIEW

File: src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java [hunk 1]
```diff
@@ -77,6 +78,12 @@ public CompletionStage<List<DataFetcherResult<Object>>> load(List<DataFetchingEn
         .thenApply(result -> batchResultTransformer.toBatchResult(result, dataFetchingEnvironments));
     }
 
+    private Map<String, Object> getSource(DataFetchingEnvironment dataFetchingEnvironment) {
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java:81]: like it.

File: src/test/groovy/com/intuit/graphql/orchestrator/integration/federation/FederatedQueryWithKeyAliasedSimpleSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            id: String @external
+            firstName: String!
+            lastName: String!
+        }      
+    '''
+
+    def setup() {
+    }
+
+    def "test Query"() {
```
Comment (CNAChino) [inline src/test/groovy/com/intuit/graphql/orchestrator/integration/federation/FederatedQueryWithKeyAliasedSimpleSpec.groovy:53]: Test should now be passing.  Will refactor

**Comment by CNAChino** [comment]
Still need to add more test cases

File: src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java [hunk 3]
```diff
@@ -133,20 +133,43 @@ private InlineFragment createEntityRequestInlineFragment(DataFetchingEnvironment
...
     ){
+        Map<String, Object> dataSource = dataFetchingEnvironment.getSource();
+        Map<String, String> fieldToAliasMap = buildFieldToAliasMap(dataFetchingEnvironment);
+
         Map<String, Object> entityRepresentation = new HashMap<>();
         entityRepresentation.put(Introspection.TypeNameMetaFieldDef.getName(), this.entityTypeName);
 
         this.representationFieldTemplate
-                .forEach(fieldName -> entityRepresentation.put(fieldName, dataSource.get(fieldName)));
+            .forEach(fieldName -> {
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java:152]: can we do this as part of the `buildFieldToAliasMap` function ?

**Review by ashpak-shaikh — APPROVED** [review]
Approach LGTM. Minor comments

File: src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java (hunk 3 above)
Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java:152]: looks like we can't combine to `buildFieldToAliasMap`.  
- this loop builds the representation.  While buildFieldToAliasMap converts SelectionSet of key - alias mapping.  Keys are optional in SelectionSet so need loop representationFieldTemplate in any case.

File: src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java [hunk 4]
```diff
@@ -133,20 +133,43 @@ private InlineFragment createEntityRequestInlineFragment(DataFetchingEnvironment
...
+     * builds mapping of fieldName-alias.  If fieldName has no alias, it will be mapped to itself.
+     *
+     * @param dataFetchingEnvironment
+     * @return
+     */
+    private Map<String, String> buildFieldToAliasMap(DataFetchingEnvironment dataFetchingEnvironment) {
+        MergedField parentField = dataFetchingEnvironment.getExecutionStepInfo().getParent().getField();
+        return parentField.getSingleField().getSelectionSet().getSelections()
+            .stream()
+            .filter(selection -> selection instanceof Field)
```
[the remark this answers is not in the document]
Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java:170]: Will try to support.. if it gets complicated will resolve in a diffrent Issue

File: src/test/groovy/com/intuit/graphql/orchestrator/integration/federation/FederatedQueryWithEntityKeyAliasedSimpleCaseSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,125 @@
+package com.intuit.graphql.orchestrator.integration.federation
```
Comment (CNAChino) [inline src/test/groovy/com/intuit/graphql/orchestrator/integration/federation/FederatedQueryWithEntityKeyAliasedSimpleCaseSpec.groovy:1]: refactoring to be  parameterized  test

**Review by ashpak-shaikh — APPROVED** [review]

**Pull request merged**